tools/filtering_by_cgroups.txt

```diff
@@ -0,0 +1,53 @@
+Demonstrations of filtering by cgroups
+
+Some tools have an option to filter by cgroup by referencing a pinned BPF hash
+map managed externally.
+
+Examples of commands:
+
+# ./opensnoop -c /sys/fs/bpf/test01
+# ./execsnoop -c /sys/fs/bpf/test01
+
+The commands above will only display results from processes that belong to one
+of the cgroups whose id, returned by bpf_get_current_cgroup_id(), is in the
+pinned BPF hash map.
+
+The BPF hash map can be created by:
+
+# bpftool map create /sys/fs/bpf/test01 type hash key 8 value 8 entries 128 \
+        name cgroupset flags 0
+
+To get a shell in a new cgroup, you can use:
+
+# systemd-run --pty --unit testbcc bash
+
+The shell will be running in the cgroup
+/sys/fs/cgroup/unified/system.slice/testbcc.service
+
+The cgroup id can be discovered using the name_to_handle_at() system call. In
+the examples/cgroupid, you will find an example of program to get the cgroup
+id.
+
+# cd examples/cgroupid
+# docker build -t cgroupid .
+# CGROUP=/sys/fs/cgroup/unified/system.slice/testbcc.service
+# CGROUPID=$(docker run -ti --rm \
+	--privileged \
+	-v /sys/fs/cgroup:/sys/fs/cgroup \
+	cgroupid \
+	cgroupid $CGROUP | tr -d '\r')
+
+Once you discovered the cgroup id, it can be added in the BPF hash map in the
+following way:
+
+# FILE=/sys/fs/bpf/test01
+# CGROUPID_HEX=$(printf %016x $CGROUPID | sed -e 's/\(..\)/0x\1\n/g' | tac)
```

Comment:
I think we could add an option in `examples/cgroupid/cgroupid.c` to display the number in the format expected by bpftool (with host endianness), instead of using `sed` + `tac` in the documentation.

Reply:
That will be good. People might be confused what to put for `value hex ...` stuff. More clarify is better.

Reply:
Done